### PR TITLE
Propagate BasePath to subdirectory storages for consistency

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Platform/Storage.cs [hunk 1]
```diff
@@ -4,14 +4,32 @@
...
+
+                if (dependentStorages != null)
+                {
+                    foreach (var s in dependentStorages)
+                        s.BasePath = basePath;
+                }
+            }
+        }
+
+        private WeakList<Storage> dependentStorages;
```
Comment (iiSaLMaN) [inline osu.Framework/Platform/Storage.cs:32]: private fields should probably be at the top

Comment (peppy) [reply osu.Framework/Platform/Storage.cs:32]: such minor discrepancies in ordering are pretty pointless to point out at this point.

**Comment by peppy** [comment]
Hold off merging this one as I reassess https://github.com/ppy/osu/pull/8935 please. Looking at whether we can potentially seal `Storage` or whether a completely different solution is required.

**Comment by smoogipoo** [comment]
Closing as this was superseded in the referencing PR.

**Pull request closed without merging**